### PR TITLE
Changes for testQosSaiPgSharedWatermark on cisco-8000 platform

### PR DESCRIPTION
**Pull request by jsanghra**
### Description of PR
Enables testQosSaiPgSharedWatermark on Cisco-8000. 

Test case modifications for Cisco-8000:
- Account for leakout using utility.
- Initial number of packets sent decreased to 1 without a margin. 
- Add asic_type to sai_thrift_read_pg_shared_watermark.
- Return occupancy instead of watermark for sai_thrift_read_pg_shared_watermark.

#### What is the motivation for this PR?
Provide Cisco-8000 support for testQosSaiPgSharedWatermark. 
#### How did you verify/test it?
Tested the Changes on Local setup and testcase passes .

### CODE REVIEW

**Comment by kevinskwang** [comment]
I believe this PR depends on #5512 , it can't be merged until 5512 is merged. You'd better just create a separate commit on #5512. That should make more sense.

File: tests/saitests/switch.py [hunk 1]
```diff
@@ -769,10 +769,11 @@ def sai_thrift_read_pg_drop_counters(client, port_id):
 
     return pg_cntrs
 
-def sai_thrift_read_pg_shared_watermark(client, port_id):
-    pg_cntr_ids=[
-        SAI_INGRESS_PRIORITY_GROUP_STAT_SHARED_WATERMARK_BYTES
-    ]
+def sai_thrift_read_pg_shared_watermark(client, asic_type, port_id):
+    if 'cisco-8000' in asic_type:
+        pg_cntr_ids = [SAI_INGRESS_PRIORITY_GROUP_STAT_CURR_OCCUPANCY_BYTES]
```
Comment (kevinskwang) [inline tests/saitests/switch.py:774]: So looks like it is just the current queue stats? If so, how it works as watermark?

**Comment by kevinskwang** [comment]
Could you address the conflicts? @jsanghra

**Comment by jsanghra** [comment]
Hi Kevin , Resolved the merge conflicts

**Review by kevinskwang — APPROVED** [review]

**Pull request merged**